jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -1815,6 +1817,18 @@ public void setMetaData(org.eclipse.jetty.http.MetaData.Request request)
 
         setMethod(request.getMethod());
         HttpURI uri = request.getURI();
+
+        if (uri.hasAmbiguousSegment())
+        {
+            // TODO replace in jetty-10 with HttpCompliance from the HttpConfiguration
+            Connection connection = _channel.getConnection();
+            boolean allow = connection instanceof HttpConnection
+                ? ((HttpConnection)connection).getHttpCompliance().sections().contains(HttpComplianceSection.AMBIGUOUS_PATH_SEGMENTS)
+                : _channel.getServer().getAttribute(HttpComplianceSection.AMBIGUOUS_PATH_SEGMENTS.toString()) == Boolean.TRUE;
+            if (!allow)
+                throw new BadMessageException("Ambiguous segment in URI");
+        }
+
```

Comment:
No tests for this code?

Reply:
Test added. I also inverted the sense of the compliance to better align with others.